libpod/oci_conmon_linux.go

```diff
@@ -0,0 +1,1402 @@
+// +build linux
+
+package libpod
```

Comment:
I wonder if we can (or should?) move this into a dedicated package `github.com/containers/libpod/libpod/conmon` or create a new `c/libpod/runtimes/` defining the interface and maybe some utility functions and place implementations as a sub-package (e.g., `c/libpod/runtimes/conmon`). I prefer the latter. This would then help to decouple the interfaces and packages.

The number of SLOCs in libpod is growing high and this could be an occasion to untangle a few bits.

Reply:
I definitely want to move this out of `libpod` into something like `libpod/oci/` but that is going to be another massive set of changes - we'll no longer be able to directly access container fields, so we'll end up needing to adjust arguments and returns a lot.

@haircommander is investigating merging our OCI code with CRI-O's and I think moving it out into a separate package helps that even more.

Reply:
+1 moving this into separate package is the second step after thing

Reply:
TBH I'd really like the oci package to live in `containers/conmon`, or at least the runtimeConmonOCI structure and definition

Reply:
I don't know about moving Go code into containers/conmon... We have containers/common now for shared Go code, so that could be a good place, and both repos can refer to it there?

Reply:
I've already started a bit by adding the `runner` package. I am curious your hesitation about adding go code there. From a code organization standpoint, it makes most sense to me to have the direct user of conmon live as close as possible, so configuration and shared changes can be made together clearly

Reply:
Hmm. I thought we'd want to keep that repo simple and C only. If we're already committed to landing Go there I'm fine with moving Conmon specific things into the Conmon repo.